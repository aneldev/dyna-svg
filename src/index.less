.dyna-svg-defaults {
	width: 63.4477px;
}
