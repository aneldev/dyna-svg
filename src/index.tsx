import * as React from "react";

import "./index.less";

export interface IDynaSvgProps extends React.HTMLAttributes<HTMLDivElement> {
  className?: string;
  src: string;
  [propsName: string]: any;
}

export class DynaSvg extends React.Component<IDynaSvgProps, {}> {
  public render(): JSX.Element {
    const {className, src, style} = this.props;
    let props = {...this.props};
    delete props.className;
    delete props.src;

    return (
      <div
        className={className || 'dyna-svg-defaults'}
        {...props}
        style={{
          ...(style || {}),
          display: 'inline-block',
          fontSize: 0,
        }}
        dangerouslySetInnerHTML={{__html: src}}
      />
    );
  }
}
