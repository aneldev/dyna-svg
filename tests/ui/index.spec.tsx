import "jest";

import { configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import * as React from 'react';
import * as enzyme from 'enzyme';

import {DynaSvg} from '../../src';

const svgIcon: string = require('./assets/svg-file-format-variant.svg');


describe('Home', () => {
  let wrapper;

  it('has expected content with deep render', () => {
    wrapper = enzyme.shallow(
      (
        <DynaSvg src={svgIcon} style={{width: "20px"}}/>
      ),
      {}
    );

    expect(wrapper).toMatchSnapshot();
  });
});
