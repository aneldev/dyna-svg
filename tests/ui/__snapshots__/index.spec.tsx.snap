// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home has expected content with deep render 1`] = `
<div
  className="dyna-svg-defaults"
  dangerouslySetInnerHTML={
    Object {
      "__html": null,
    }
  }
  style={
    Object {
      "display": "inline-block",
      "fontSize": 0,
      "width": "20px",
    }
  }
/>
`;
